x-pack/plugins/rule_registry/common/types.ts

```diff
@@ -7,6 +7,270 @@
+              max: t.number,
+            })
+          ),
+          missing: t.number,
+          keyed: t.boolean,
+          order: t.exact(
+            t.type({
+              _count: t.string,
+              _key: t.string,
+            })
+          ),
+        })
+      ),
+      nested: t.type({
+        path: t.string,
+      }),
+      terms: t.exact(
+        t.partial({
+          field: t.string,
+          collect_mode: t.string,
+          exclude: t.union([t.string, t.array(t.string)]),
+          include: t.union([t.string, t.array(t.string)]),
+          execution_hint: t.string,
+          missing: t.number,
+          min_doc_count: t.number,
+          size: t.number,
+          show_term_doc_count_error: t.boolean,
+          order: t.union([
+            sortOrderSchema,
+            t.record(t.string, sortOrderSchema),
+            t.array(t.record(t.string, sortOrderSchema)),
+          ]),
+        })
+      ),
+      aggs: t.record(t.string, BucketAggsSchemas),
+    })
+  )
+);
+
+/**
+ * Schemas for the metrics Aggregations
+ *
+ * Currently supported:
+ * - avg
+ * - cardinality
+ * - min
+ * - max
+ * - sum
+ * - top_hits
+ * - weighted_avg
+ *
+ * Not implemented:
+ * - boxplot
+ * - extended_stats
+ * - geo_bounds
+ * - geo_centroid
+ * - geo_line
+ * - matrix_stats
+ * - median_absolute_deviation
+ * - percentile_ranks
+ * - percentiles
+ * - rate
+ * - scripted_metric
+ * - stats
+ * - string_stats
+ * - t_test
+ * - value_count
+ */
+export const metricsAggsSchemas = t.partial({
+  avg: t.partial({
+    field: t.string,
+    missing: t.union([t.string, t.number, t.boolean]),
+  }),
+  cardinality: t.partial({
+    field: t.string,
+    precision_threshold: t.number,
+    rehash: t.boolean,
+    missing: t.union([t.string, t.number, t.boolean]),
+  }),
+  min: t.partial({
+    field: t.string,
+    missing: t.union([t.string, t.number, t.boolean]),
+    format: t.string,
+  }),
+  max: t.partial({
+    field: t.string,
+    missing: t.union([t.string, t.number, t.boolean]),
+    format: t.string,
+  }),
+  sum: t.partial({
+    field: t.string,
+    missing: t.union([t.string, t.number, t.boolean]),
+  }),
+  top_hits: t.partial({
+    explain: t.boolean,
+    docvalue_fields: t.union([t.string, t.array(t.string)]),
+    stored_fields: t.union([t.string, t.array(t.string)]),
+    from: t.number,
+    size: t.number,
+    sort: sortSchema,
+    seq_no_primary_term: t.boolean,
+    version: t.boolean,
+    track_scores: t.boolean,
+    highlight: t.any,
+    _source: t.union([t.boolean, t.string, t.array(t.string)]),
+  }),
+  weighted_avg: t.partial({
+    format: t.string,
+    value_type: t.string,
+    value: t.partial({
+      field: t.string,
+      missing: t.number,
+    }),
+    weight: t.partial({
+      field: t.string,
+      missing: t.number,
+    }),
+  }),
+});
+
+export type PutIndexTemplateRequest = estypes.IndicesPutIndexTemplateRequest & {
+  body?: { composed_of?: string[] };
+};
```

Comment:
Can be deleted, `estypes.IndicesPutIndexTemplateRequest` contains `composed_of`:

```ts
export interface IndicesPutIndexTemplateRequest extends RequestBase {
  name: Name
  body?: {
    index_patterns?: Indices
    composed_of?: Name[]
    template?: IndicesPutIndexTemplateIndexTemplateMapping
    data_stream?: EmptyObject
    priority?: integer
    version?: VersionNumber
    _meta?: Metadata
  }
}
```

Reply:
ah good catch. I copied some of these types from elsewhere in kibana so didn't look too deeply at the types but I will update this. Thanks!